src/Processors/Formats/Impl/ParquetBlockInputFormat.cpp

```diff
@@ -656,6 +688,29 @@ void ParquetBlockInputFormat::threadFunction(size_t row_group_batch_idx)
             return;
     }
 }
+bool ParquetBlockInputFormat::supportPrefetch() const
+{
+    return max_decoding_threads == 1 && max_io_threads > 0 && format_settings.parquet.enable_row_group_prefetch && !format_settings.parquet.use_native_reader;
+}
+
+std::shared_ptr<arrow::RecordBatchReader> ParquetBlockInputFormat::RowGroupPrefetchIterator::nextRowGroupReader()
+{
+    if (next_row_group_idx >= row_group_batch.row_groups_idxs.size()) return nullptr;
+    std::shared_ptr<arrow::RecordBatchReader> reader;
+    THROW_ARROW_NOT_OK(row_group_batch.file_reader->GetRecordBatchReader({row_group_batch.row_groups_idxs[next_row_group_idx]}, column_indices, &reader));
+    prefetchNextRowGroupReader();
```

Comment:
could you clarify pls: do we prefetch group `g` before starting to read it or do we prefetch group `g+1` before starting to read group `g`?